# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/fetch/FetchUtil.cpp
# Commit: 3dd08e02c0c2
# Full Hash: 3dd08e02c0c2a8a048c2f148492b2a167e3711ec
# Author: Yury Delendik <ydelendik@mozilla.com>
# Date: 2021-11-01 21:59:26
# Description:
#   Bug 1737837 - Fix truncated storeOptimizedEncoding stream. r=jandem
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D129880
# ==============================================================================

diff -r d8ee93e73486 -r 3dd08e02c0c2 dom/fetch/FetchUtil.cpp
--- a/dom/fetch/FetchUtil.cpp	Mon Nov 01 13:37:38 2021 +0000
+++ b/dom/fetch/FetchUtil.cpp	Mon Nov 01 13:38:40 2021 +0000
@@ -627,7 +627,7 @@
     }
     MOZ_RELEASE_ASSERT(ret == Z_STREAM_END);
 
-    dstBytes.shrinkTo(zstream.next_out - compressBegin);
+    dstBytes.shrinkTo(zstream.next_out - dstBytes.begin());
 
     NS_DispatchToMainThread(new StoreOptimizedEncodingRunnable(
         std::move(mCache), std::move(dstBytes)));
